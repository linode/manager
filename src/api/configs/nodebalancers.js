import {
  genConfig, ReducerGenerator, genActions,
  ONE, MANY, PUT, DELETE, POST,
} from '~/api/apiResultActionReducerGenerator';

export const config = genConfig({
  plural: 'nodebalancers',
  singular: 'nodebalancers',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [ONE, MANY, PUT, DELETE, POST],
  subresources: {
    _configs: {
      plural: 'configs',
      singular: 'config',
      endpoint: (id, nbConfigId) => `/nodebalancers/${id}/configs/${nbConfigId}`,
      supports: [ONE, MANY, PUT, POST, DELETE],
    },
  },


});

export const actions = genActions(config);
export const { reducer } = new ReducerGenerator(config);
