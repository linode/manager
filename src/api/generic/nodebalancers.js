import {
  genConfig, ReducerGenerator, genActions,
  ONE, MANY, PUT, DELETE, POST,
} from '~/api/internal';

export const config = genConfig({
  plural: 'nodebalancers',
  endpoint: id => `/nodebalancers/${id}`,
  supports: [ONE, MANY, PUT, DELETE, POST],
  subresources: {
    _generic: {
      plural: 'generic',
      endpoint: (id, nbConfigId) => `/nodebalancers/${id}/generic/${nbConfigId}`,
      supports: [ONE, MANY, PUT, POST, DELETE],
      subresources: {
        _nodes: {
          plural: 'nodes',
          endpoint: (id, nbConfigId, nodeId) => {
            return `/nodebalancers/${id}/generic/${nbConfigId}/nodes/${nodeId}`;
          },
          supports: [ONE, MANY, PUT, POST, DELETE],
        },
      },
    },
  },


});

export const actions = genActions(config);
export const { reducer } = new ReducerGenerator(config);
