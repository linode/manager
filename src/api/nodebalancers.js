import { thunkFetch } from './apiActionReducerGenerator';

export function updateConfigSSL(data, nodebalancerId, configId) {
  return thunkFetch.post(`/nodebalancers/${nodebalancerId}/configs/${configId}/ssl`, data);
}
