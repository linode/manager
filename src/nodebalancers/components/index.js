export { default as AddNodeBalancer } from './AddNodeBalancer';
