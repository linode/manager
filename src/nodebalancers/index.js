import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import NodeBalancersList from './layouts/NodeBalancersList';
import NodeBalancerIndex from './nodebalancer';
import Configs from './nodebalancer/configs';
import NodeBalancerConfigAdd from './nodebalancer/layouts/NodeBalancerConfigAdd';

const NodeBalancersIndex = (props) => {
  const { match: { path } } = props;
  return (
    <Switch>
      <Route component={NodeBalancerConfigAdd} path={`${path}/:nbLabel/configs/create`} />
      <Route component={Configs} path={`${path}/:nbLabel/configs/:configId`} />
      <Route component={NodeBalancerIndex} path={`${path}/:nbLabel`} />
      <Route component={NodeBalancersList} exact path={`${path}/`} />
      <Redirect to="/not-found" />
    </Switch >
  );
};

NodeBalancersIndex.propTypes = {
  match: PropTypes.shape({
    path: PropTypes.string.isRequired,
  }).isRequired,
};

export default NodeBalancersIndex;
