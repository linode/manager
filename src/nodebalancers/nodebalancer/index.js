import React from 'react';
import { Route, IndexRoute } from 'react-router';

import IndexPage from './layouts/IndexPage';
import DashboardPage from './layouts/DashboardPage';
import EditConfigPage from './layouts/EditConfigPage';
import AddConfigPage from './layouts/AddConfigPage';
import SettingsPage from './layouts/SettingsPage';

export default (
  <Route path=":nbLabel" component={IndexPage}>
    <IndexRoute component={DashboardPage} />
    <Route path="settings" component={SettingsPage} />
    <Route path="configurations/create" component={AddConfigPage} />
    <Route path="configurations/:configId" component={EditConfigPage} />
  </Route>
);
