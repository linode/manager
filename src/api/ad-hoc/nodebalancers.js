import { actions } from '../generic/nodebalancers';
import { fetch } from '../fetch';

export function nodebalancerStats(nodebalancerId) {
  return async (dispatch) => {
    const { data: _stats } = await dispatch(
      fetch.get(`/nodebalancers/${nodebalancerId}/stats`)
    );
    dispatch(actions.one({ _stats }, nodebalancerId));
  };
}
