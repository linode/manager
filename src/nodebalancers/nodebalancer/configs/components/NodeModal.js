import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { ModalFormGroup, Input, Select } from 'linode-components';
import { onChange } from 'linode-components';
import { FormModalBody } from 'linode-components';

import { showModal } from '~/actions/modal';
import api from '~/api';
import { dispatchOrStoreErrors } from '~/api/util';


export default class NodeModal extends Component {
  // For some reason NodeModal.trigger only recognizes when the defaultProps
  // are listed like this.
  static defaultProps = {
    node: {
      id: '',
      label: '',
      address: '',
      weight: '',
      mode: 'accept',
    },
  }

  static trigger(dispatch, nodebalancer, config, node) {
    const title = node ? 'Edit Node' : 'Add a Node';

    dispatch(showModal(title, (
      <NodeModal
        dispatch={dispatch}
        configId={config.id}
        nodebalancerId={nodebalancer.id}
        title={title}
        node={node}
      />
    )));
  }

  constructor(props) {
    super(props);

    this.state = {
      id: props.node && props.node.id,
      label: props.node && props.node.label,
      address: props.node && props.node.address,
      weight: props.node && props.node.weight,
      mode: props.node && props.node.mode,
      errors: {},
    };

    this.onChange = onChange.bind(this);
  }

  onSubmit = () => {
    const { dispatch, nodebalancerId, configId, close } = this.props;
    const state = this.state;
    const data = {
      label: state.label,
      address: state.address,
      weight: +state.weight,
      mode: state.mode,
    };
    const ids = [nodebalancerId, configId, state.id].filter(Boolean);

    return dispatch(dispatchOrStoreErrors.call(this, [
      () => api.nodebalancers.configs.nodes[state.id ? 'put' : 'post'](data, ...ids),
      close,
    ]));
  }

  render() {
    const { title, close } = this.props;
    const { id, label, address, weight, mode, errors } = this.state;

    const modeOptions = [
      { value: 'accept', label: 'Accept' },
      { value: 'reject', label: 'Reject' },
      { value: 'drain', label: 'Drain' },
    ];

    return (
      <FormModalBody
        onSubmit={this.onSubmit}
        onCancel={close}
        buttonText={id ? undefined : 'Add Node'}
        buttonDisabledText={id ? undefined : 'Adding Node'}
        analytics={{ title, action: id ? 'edit' : 'add' }}
        errors={errors}
      >
        <div>
          <ModalFormGroup
            label="Label"
            id="label"
            apiKey="label"
            errors={errors}
          >
            <Input
              name="label"
              id="label"
              onChange={this.onChange}
              value={label}
            />
          </ModalFormGroup>
          <ModalFormGroup
            label="Address"
            id="address"
            apiKey="address"
            errors={errors}
          >
            <Input
              name="address"
              id="address"
              placeholder="192.168.1.10:80"
              onChange={this.onChange}
              value={address}
            />
          </ModalFormGroup>
          <ModalFormGroup
            label="Weight"
            description="Define a weight to be used in determining how
                         connections are balanced to this node."
            id="weight"
            apiKey="weight"
            errors={errors}
          >
            <Input
              name="weight"
              id="weight"
              type="number"
              onChange={this.onChange}
              value={weight}
            />
          </ModalFormGroup>
          <ModalFormGroup
            label="Mode"
            description="Configure how this node handles incoming connections
                         based on it's health."
            id="mode"
            apiKey="mode"
            errors={errors}
          >
            <Select
              id="mode"
              name="mode"
              onChange={this.onChange}
              value={mode}
              options={modeOptions}
            />
          </ModalFormGroup>
        </div>
      </FormModalBody>
    );
  }
}

NodeModal.propTypes = {
  dispatch: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  configId: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  nodebalancerId: PropTypes.number.isRequired,
  node: PropTypes.object,
  close: PropTypes.func.isRequired,
};
