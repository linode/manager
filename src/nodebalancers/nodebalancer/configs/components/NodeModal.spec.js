import React from 'react';
import sinon from 'sinon';
import { mount, shallow } from 'enzyme';

import {
  createSimulatedEvent,
  expectDispatchOrStoreErrors,
  expectRequest,
} from '~/test.helpers';
import { configsNodeBalancer } from '~/data/nodebalancers';
import NodeModal from '~/nodebalancers/nodebalancer/configs/components/NodeModal';

const node = configsNodeBalancer._configs.configs['1']._nodes.nodes[1];

describe('nodebalancers/nodebalancer/configs/components/NodeModal', () => {
  const sandbox = sinon.sandbox.create();

  afterEach(() => {
    sandbox.restore();
  });

  const dispatch = sandbox.stub();

  it('should render without error', () => {
    const dispatch = jest.fn();
    const wrapper = shallow(
      <NodeModal
        title="NodeModal"
        dispatch={dispatch}
        confirmTest="Edit"
        configId="1"
        nodebalancerId="23"
        node={node}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders node modal with data', () => {
    const page = mount(
      <NodeModal
        title="NodeModal"
        dispatch={dispatch}
        confirmTest="Edit"
        configId="1"
        nodebalancerId="23"
        node={node}
      />
    );

    expect(page.find('input#label').props().value).toBe('greatest_node_ever');
    expect(page.find('input#address').props().value).toBe('192.168.4.5:80');
    expect(page.find('input#weight').props().value).toBe(40);
    expect(page.find('select#mode').props().value).toBe('accept');
  });

  it.skip('updates node', async () => {
    const page = mount(
      <NodeModal
        title="NodeModal"
        dispatch={dispatch}
        confirmTest="Edit"
        configId="1"
        nodebalancerId="23"
        node={node}
      />
    );

    dispatch.reset();
    await page.find('Form').props().onSubmit();
    expect(dispatch.callCount).toBe(1);
    await expectDispatchOrStoreErrors(dispatch.firstCall.args[0], [
      ([fn]) => expectRequest(fn, `/nodebalancers/23/configs/1/nodes/${node.id}`, {
        method: 'PUT',
        body: {
          label: 'greatest_node_ever',
          address: '192.168.4.5:80',
          weight: 40,
          mode: 'accept',
        },
      }),
    ], 2);
  });

  it.skip('creates node', async () => {
    const page = mount(
      <NodeModal
        title="NodeModal"
        dispatch={dispatch}
        confirmTest="Edit"
        configId="1"
        nodebalancerId="23"
      />
    );

    page.find('input#label')
      .simulate('change', createSimulatedEvent('label', 'myLabel'));
    page.find('input#address')
      .simulate('change', createSimulatedEvent('address', '192.168.4.6:88'));
    page.find('input#weight')
      .simulate('change', createSimulatedEvent('weight', '50'));

    dispatch.reset();
    await page.find('Form').props().onSubmit();
    expect(dispatch.callCount).toBe(1);
    await expectDispatchOrStoreErrors(dispatch.firstCall.args[0], [
      ([fn]) => expectRequest(fn, '/nodebalancers/23/configs/1/nodes/', {
        method: 'POST',
        body: {
          label: 'myLabel',
          address: '192.168.4.6:88',
          weight: 50,
          mode: 'accept',
        },
      }),
    ], 2);
  });
});
