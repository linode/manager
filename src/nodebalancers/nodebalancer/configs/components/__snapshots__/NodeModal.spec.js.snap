// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodebalancers/nodebalancer/configs/components/NodeModal should render without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NodeModal
    configId="1"
    confirmTest="Edit"
    dispatch={[Function]}
    node={
      Object {
        "address": "192.168.4.5:80",
        "config_id": 0,
        "id": 1,
        "label": "greatest_node_ever",
        "mode": "accept",
        "nodebalancer_id": 23,
        "status": "online",
        "weight": 40,
      }
    }
    nodebalancerId="23"
    title="NodeModal"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "analytics": Object {
        "action": "edit",
        "title": "NodeModal",
      },
      "buttonDisabledText": "Saving",
      "buttonText": "Save",
      "children": <div>
        <ModalFormGroup
          apiKey="label"
          errors={Object {}}
          id="label"
          label="Label"
        >
          <Input
            className=""
            id="label"
            name="label"
            onChange={[Function]}
            value="greatest_node_ever"
          />
        </ModalFormGroup>
        <ModalFormGroup
          apiKey="address"
          errors={Object {}}
          id="address"
          label="Address"
        >
          <Input
            className=""
            id="address"
            name="address"
            onChange={[Function]}
            placeholder="192.168.1.10:80"
            value="192.168.4.5:80"
          />
        </ModalFormGroup>
        <ModalFormGroup
          apiKey="weight"
          description="Define a weight to be used in determining how connections are balanced to this node."
          errors={Object {}}
          id="weight"
          label="Weight"
        >
          <Input
            className=""
            id="weight"
            name="weight"
            onChange={[Function]}
            type="number"
            value={40}
          />
        </ModalFormGroup>
        <ModalFormGroup
          apiKey="mode"
          description="Configure how this node handles incoming connections based on it's health."
          errors={Object {}}
          id="mode"
          label="Mode"
        >
          <Select
            analytics={
              Object {
                "noTrack": false,
              }
            }
            className=""
            id="mode"
            name="mode"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "label": "Accept",
                  "value": "accept",
                },
                Object {
                  "label": "Reject",
                  "value": "reject",
                },
                Object {
                  "label": "Drain",
                  "value": "drain",
                },
              ]
            }
            value="accept"
          />
        </ModalFormGroup>
      </div>,
      "errors": Object {},
      "noCancel": false,
      "noCancelEvent": false,
      "noSubmit": false,
      "onCancel": undefined,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ModalFormGroup
            apiKey="label"
            errors={Object {}}
            id="label"
            label="Label"
          >
            <Input
              className=""
              id="label"
              name="label"
              onChange={[Function]}
              value="greatest_node_ever"
            />
          </ModalFormGroup>,
          <ModalFormGroup
            apiKey="address"
            errors={Object {}}
            id="address"
            label="Address"
          >
            <Input
              className=""
              id="address"
              name="address"
              onChange={[Function]}
              placeholder="192.168.1.10:80"
              value="192.168.4.5:80"
            />
          </ModalFormGroup>,
          <ModalFormGroup
            apiKey="weight"
            description="Define a weight to be used in determining how connections are balanced to this node."
            errors={Object {}}
            id="weight"
            label="Weight"
          >
            <Input
              className=""
              id="weight"
              name="weight"
              onChange={[Function]}
              type="number"
              value={40}
            />
          </ModalFormGroup>,
          <ModalFormGroup
            apiKey="mode"
            description="Configure how this node handles incoming connections based on it's health."
            errors={Object {}}
            id="mode"
            label="Mode"
          >
            <Select
              analytics={
                Object {
                  "noTrack": false,
                }
              }
              className=""
              id="mode"
              name="mode"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Accept",
                    "value": "accept",
                  },
                  Object {
                    "label": "Reject",
                    "value": "reject",
                  },
                  Object {
                    "label": "Drain",
                    "value": "drain",
                  },
                ]
              }
              value="accept"
            />
          </ModalFormGroup>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "apiKey": "label",
            "children": <Input
              className=""
              id="label"
              name="label"
              onChange={[Function]}
              value="greatest_node_ever"
            />,
            "errors": Object {},
            "id": "label",
            "label": "Label",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "id": "label",
              "name": "label",
              "onChange": [Function],
              "value": "greatest_node_ever",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "apiKey": "address",
            "children": <Input
              className=""
              id="address"
              name="address"
              onChange={[Function]}
              placeholder="192.168.1.10:80"
              value="192.168.4.5:80"
            />,
            "errors": Object {},
            "id": "address",
            "label": "Address",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "id": "address",
              "name": "address",
              "onChange": [Function],
              "placeholder": "192.168.1.10:80",
              "value": "192.168.4.5:80",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "apiKey": "weight",
            "children": <Input
              className=""
              id="weight"
              name="weight"
              onChange={[Function]}
              type="number"
              value={40}
            />,
            "description": "Define a weight to be used in determining how connections are balanced to this node.",
            "errors": Object {},
            "id": "weight",
            "label": "Weight",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "id": "weight",
              "name": "weight",
              "onChange": [Function],
              "type": "number",
              "value": 40,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "apiKey": "mode",
            "children": <Select
              analytics={
                Object {
                  "noTrack": false,
                }
              }
              className=""
              id="mode"
              name="mode"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Accept",
                    "value": "accept",
                  },
                  Object {
                    "label": "Reject",
                    "value": "reject",
                  },
                  Object {
                    "label": "Drain",
                    "value": "drain",
                  },
                ]
              }
              value="accept"
            />,
            "description": "Configure how this node handles incoming connections based on it's health.",
            "errors": Object {},
            "id": "mode",
            "label": "Mode",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "analytics": Object {
                "noTrack": false,
              },
              "className": "",
              "id": "mode",
              "name": "mode",
              "onChange": [Function],
              "options": Array [
                Object {
                  "label": "Accept",
                  "value": "accept",
                },
                Object {
                  "label": "Reject",
                  "value": "reject",
                },
                Object {
                  "label": "Drain",
                  "value": "drain",
                },
              ],
              "value": "accept",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "analytics": Object {
          "action": "edit",
          "title": "NodeModal",
        },
        "buttonDisabledText": "Saving",
        "buttonText": "Save",
        "children": <div>
          <ModalFormGroup
            apiKey="label"
            errors={Object {}}
            id="label"
            label="Label"
          >
            <Input
              className=""
              id="label"
              name="label"
              onChange={[Function]}
              value="greatest_node_ever"
            />
          </ModalFormGroup>
          <ModalFormGroup
            apiKey="address"
            errors={Object {}}
            id="address"
            label="Address"
          >
            <Input
              className=""
              id="address"
              name="address"
              onChange={[Function]}
              placeholder="192.168.1.10:80"
              value="192.168.4.5:80"
            />
          </ModalFormGroup>
          <ModalFormGroup
            apiKey="weight"
            description="Define a weight to be used in determining how connections are balanced to this node."
            errors={Object {}}
            id="weight"
            label="Weight"
          >
            <Input
              className=""
              id="weight"
              name="weight"
              onChange={[Function]}
              type="number"
              value={40}
            />
          </ModalFormGroup>
          <ModalFormGroup
            apiKey="mode"
            description="Configure how this node handles incoming connections based on it's health."
            errors={Object {}}
            id="mode"
            label="Mode"
          >
            <Select
              analytics={
                Object {
                  "noTrack": false,
                }
              }
              className=""
              id="mode"
              name="mode"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "Accept",
                    "value": "accept",
                  },
                  Object {
                    "label": "Reject",
                    "value": "reject",
                  },
                  Object {
                    "label": "Drain",
                    "value": "drain",
                  },
                ]
              }
              value="accept"
            />
          </ModalFormGroup>
        </div>,
        "errors": Object {},
        "noCancel": false,
        "noCancelEvent": false,
        "noSubmit": false,
        "onCancel": undefined,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ModalFormGroup
              apiKey="label"
              errors={Object {}}
              id="label"
              label="Label"
            >
              <Input
                className=""
                id="label"
                name="label"
                onChange={[Function]}
                value="greatest_node_ever"
              />
            </ModalFormGroup>,
            <ModalFormGroup
              apiKey="address"
              errors={Object {}}
              id="address"
              label="Address"
            >
              <Input
                className=""
                id="address"
                name="address"
                onChange={[Function]}
                placeholder="192.168.1.10:80"
                value="192.168.4.5:80"
              />
            </ModalFormGroup>,
            <ModalFormGroup
              apiKey="weight"
              description="Define a weight to be used in determining how connections are balanced to this node."
              errors={Object {}}
              id="weight"
              label="Weight"
            >
              <Input
                className=""
                id="weight"
                name="weight"
                onChange={[Function]}
                type="number"
                value={40}
              />
            </ModalFormGroup>,
            <ModalFormGroup
              apiKey="mode"
              description="Configure how this node handles incoming connections based on it's health."
              errors={Object {}}
              id="mode"
              label="Mode"
            >
              <Select
                analytics={
                  Object {
                    "noTrack": false,
                  }
                }
                className=""
                id="mode"
                name="mode"
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Accept",
                      "value": "accept",
                    },
                    Object {
                      "label": "Reject",
                      "value": "reject",
                    },
                    Object {
                      "label": "Drain",
                      "value": "drain",
                    },
                  ]
                }
                value="accept"
              />
            </ModalFormGroup>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "apiKey": "label",
              "children": <Input
                className=""
                id="label"
                name="label"
                onChange={[Function]}
                value="greatest_node_ever"
              />,
              "errors": Object {},
              "id": "label",
              "label": "Label",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "id": "label",
                "name": "label",
                "onChange": [Function],
                "value": "greatest_node_ever",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "apiKey": "address",
              "children": <Input
                className=""
                id="address"
                name="address"
                onChange={[Function]}
                placeholder="192.168.1.10:80"
                value="192.168.4.5:80"
              />,
              "errors": Object {},
              "id": "address",
              "label": "Address",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "id": "address",
                "name": "address",
                "onChange": [Function],
                "placeholder": "192.168.1.10:80",
                "value": "192.168.4.5:80",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "apiKey": "weight",
              "children": <Input
                className=""
                id="weight"
                name="weight"
                onChange={[Function]}
                type="number"
                value={40}
              />,
              "description": "Define a weight to be used in determining how connections are balanced to this node.",
              "errors": Object {},
              "id": "weight",
              "label": "Weight",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "",
                "id": "weight",
                "name": "weight",
                "onChange": [Function],
                "type": "number",
                "value": 40,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "apiKey": "mode",
              "children": <Select
                analytics={
                  Object {
                    "noTrack": false,
                  }
                }
                className=""
                id="mode"
                name="mode"
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "Accept",
                      "value": "accept",
                    },
                    Object {
                      "label": "Reject",
                      "value": "reject",
                    },
                    Object {
                      "label": "Drain",
                      "value": "drain",
                    },
                  ]
                }
                value="accept"
              />,
              "description": "Configure how this node handles incoming connections based on it's health.",
              "errors": Object {},
              "id": "mode",
              "label": "Mode",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "analytics": Object {
                  "noTrack": false,
                },
                "className": "",
                "id": "mode",
                "name": "mode",
                "onChange": [Function],
                "options": Array [
                  Object {
                    "label": "Accept",
                    "value": "accept",
                  },
                  Object {
                    "label": "Reject",
                    "value": "reject",
                  },
                  Object {
                    "label": "Drain",
                    "value": "drain",
                  },
                ],
                "value": "accept",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
